fem/nonlinearform.cpp

```diff
@@ -53,6 +53,22 @@ void NonlinearForm::SetEssentialBC(const Array<int> &bdr_attr_is_ess,
    }
 }
 
+void NonlinearForm::SetEssentialBCPartial(const Array<int> &bdr_attr_is_ess,
```

Comment:
What does `Partial` mean in this case? The term is ambiguous. I would opt for keeping the same name `NonlinearForm::SetEssentialBC`. The added argument `bdr_component` and class method comment make it clear that this will work different from the non component version.

Reply:
`Partial` was just the name we'd had picked back in the day. It was supposed to just signify that you're not constrained to either constraining all dofs or only 1 dof. Instead, you could constrain multiple DOFs at a time for a boundary. I really don't have strong preferences to keeping the name though.